boudicca.base/enricher/src/main/kotlin/base/boudicca/enricher/service/osm/OsmEnricher.kt

```diff
@@ -0,0 +1,193 @@
+import base.boudicca.format.ListFormat
+import base.boudicca.format.UrlUtils
+import base.boudicca.format.UrlUtils.encodeURL
+import base.boudicca.model.structured.Key
+import base.boudicca.model.structured.StructuredEvent
+import base.boudicca.model.structured.StructuredEvent.StructuredEventBuilder
+import base.boudicca.model.structured.Variant
+import base.boudicca.model.structured.VariantConstants.SOURCE_VARIANT_NAME
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.ArrayNode
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import org.slf4j.LoggerFactory
+import org.springframework.core.annotation.Order
+import org.springframework.stereotype.Service
+import java.net.URI
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+
+
+@Service
+@Order(EnricherOrderConstants.OsmEnricherOrder)
+class OsmEnricher : Enricher {
+    val sourceVariant = listOf(Variant(SOURCE_VARIANT_NAME, "OSM"))
+    val logger = LoggerFactory.getLogger(this::class.java)!!
+
+    override fun enrich(event: StructuredEvent): StructuredEvent {
+        val osmId: String? = getOrSearchOsmId(event)
+        if (osmId == null || !isValidOsmIdFormat(osmId)) {
+            return event
+        }
+        val enrichedEvent = enrichEventFromOverpass(osmId, event)
+
+        return enrichedEvent
+    }
+
+    @Suppress("detekt.LongMethod")
+    private fun enrichEventFromOverpass(
+        osmId: String,
+        event: StructuredEvent
+    ): StructuredEvent {
+        val query = "[out:json];nwr($osmId);out body center;"
+        val url = "https://overpass-api.de/api/interpreter?data=${query.encodeURL()}"
+        val request = HttpRequest.newBuilder()
+            .uri(URI.create(url))
+            .GET()
+            .header("User-Agent", Constants.USER_AGENT)
+            .build()
+        val response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString()).body()
+
+        // parse osm data into our properties
+        val builder = event.toBuilder()
+        val json = jacksonObjectMapper().readTree(response)
+        val elements = json["elements"] as ArrayNode
+        val tags = elements.first()["tags"]
+        val center = elements.first()["center"]
+
+        val currentSources = ListFormat.parseFromString(event.data[SemanticKeys.SOURCES_PROPERTY.getKey()])
+
+        builder.withProperty(
+            property = SemanticKeys.LOCATION_COORDINATES_LAT_PROPERTY,
+            value = center["lat"].asText(),
+            variants = sourceVariant
+        ).withProperty(
+            property = SemanticKeys.LOCATION_COORDINATES_LON_PROPERTY,
+            value = center["lon"].asText(),
+            variants = sourceVariant
+        ).withProperty(
+            property = SemanticKeys.SOURCES_PROPERTY,
+            value = currentSources + "Source:OSM:" + json["osm3s"]["copyright"].asText()
```

Comment:
uhm... how is the operator precedence here? does this add 2 strings to the sources or only one?

Reply:
yeah, it's as you pointed it out, have changed this